docs/_dg/Implementation.md

```diff
@@ -0,0 +1,18 @@
+<!-- markdownlint-disable-file first-line-h1 -->
+This section describes some noteworthy details on how certain features are implemented.
+
+#### Tagging an item
+The `tag` feature tags an item with with the provided tag name in FoodRem. If both the item and the tag are valid,  the item will be tagged successfully.
+
+#### How the `tag` command works
+1. The user specifies a tag name and the item index (which corresponds to the index of the item displayed on the item list on FoodRem UI).
+2. If the tag name or index is not provided, the user will be prompted to enter them correctly via an error message.
+3. The tag name is cross-referenced with the current tags in the database and an error is thrown if the tag does not exist in the database.
+4. The index is cross-referenced to the list of item displayed and an error would be thrown if the index is out of range of the list. This informs the user that the item does not exist.
+5. If the item index and tag name are both valid, the item is checked to determine whether that item contains the specified tag. If the specified tag is already contained in the item, an error would be thrown to inform user about the duplicate tag.
+6. If Step 5 completes without any exceptions, a new copy of the item is created and the tag would be added to that item. This new copy of item would replace the original item in the database.
```

Comment:
Let us follow our module's markdown coding standards and use generic list item numbers. But it's okay, this can be fixed later.